### ISSUE TITLE
Feature Request: Location/Position Support

### ISSUE THREAD

**Issue by poynting** [NaturalHistoryMuseum/pyzbar#11]
I'm currently using zbar on Python 2.7 but in hopes of supporting Python 3 I checked out your library. It was very easy to get going, but unfortunately I use the location support in the zbar library to determine the location of QR codes in the image.  I didn't see a way to do this with pyzbar. I looked into adding location support to the wrapper, but I was unable to get any location information out of libzbar itself in the python context.  I tried both function signatures I expected to work as well as XML outputs. 

Do you have any ideas on if it's possible to coax location information out of zbar in the context of this wrapper?

https://github.com/ZBar/ZBar/blob/master/python/symbol.c#L128

**Comment by WTRipper** [comment]
I would like to use this feature as well. Could offer my help if something has to be done that requires only low skill level.

**Comment by hvbargen** [comment]
I made the following modifications to the source code.
With these modifications, pyzbar returns the positions (the approximate corners of the rectangle containing the symbol, as a list of 4 (x,y)-tuples).

See the attached text file.
[modifications.txt](https://github.com/NaturalHistoryMuseum/pyzbar/files/1777405/modifications.txt)

**Comment by globophobe** [comment]
I submitted a pull request https://github.com/NaturalHistoryMuseum/pyzbar/pull/12 for this.

**Comment by hvbargen** [comment]
+1 for your pull request.

**Comment by quicklizard99** [comment]
@globophobe and @hvbargen - thanks very much for the PR and upload respectively. Apologies for the delay in responding. 

Locations would be a very nice addition. Returning the complete list of x, y pairs that describe scan locations is probably only interesting to developers of the `zbar` library andI find the following output to be unhelpful (a newcomer would most likely be baffled):

```
Decoded(
    data=b'Rana temporaria',
    type='CODE128',
    location=[
        (394, 0), (4, 1), (394, 2), (4, 3),
        (394, 4), (4, 5), (394, 6), (4, 7),
        (394, 8), (4, 9), (394, 10), (4, 11),
        (394, 12), (4, 13), (394, 14), (4, 15),
        (394, 16), (4, 17), (394, 18), (4, 19),
        (394, 20), (4, 21), (394, 22), (4, 23),
        (394, 24), (4, 25), (394, 26), (4, 27),
        (394, 28), (4, 29), (394, 30), (4, 31),
        (394, 32), (4, 33), (394, 34), (4, 35),
        (394, 36), (4, 37), (394, 38), (4, 39),
        (394, 40), (4, 41), (394, 42), (4, 43),
        (394, 44), (4, 45), (394, 46), (4, 47),
        (394, 48), (4, 49), (394, 50), (4, 51),
        (394, 52), (4, 53), (394, 54), (4, 55),
        (394, 56), (4, 57), (394, 58), (4, 59),
        (394, 60), (4, 61), (394, 62), (4, 63),
        (394, 64), (4, 65), (394, 66), (4, 67),
        (394, 68), (4, 69), (394, 70), (4, 71),
        (394, 72), (4, 73), (394, 74), (4, 75),
        (394, 76)
    ]
)
```

I would prefer the default behaviour of the `decode` function to return locations as simple bounding boxes around each barcode (as with the related [pylibdmtx](https://github.com/NaturalHistoryMuseum/pylibdmtx#example-usage) library). I will comment on the PR with suggested about how to implement this. We could also make the list of x, y locations available via a different interface, should anybody find these useful.

**Comment by hvbargen** [comment]
You are right, actually in my code where I store the location as metadata, I collect min and max of the x and y values to construct a (more or less) bounding rect.

**Comment by quicklizard99** [comment]
`0.1.5`, including barcode locations, now on PyPI. I will close this issue.

Please open a new issue if you would like to see the raw locations returned by zbar.

**Comment by poynting** [comment]
Thanks @quicklizard99 for your effort on this. I apologize that I was a no-show when this was discussed, I didn't realize I wasn't getting email updates on this issue so I missed the progress! For my application I need to determine the actual corners of a QR, not just the bounding box, so I will open another issue for the raw locations; however, I'll first check out your modifications first to see if I can tackle that myself based on the work done.

**Comment by quicklizard99** [comment]
Argh, sorry that I released so quickly. How does the following sound?

Retain the existing`rect` field (which I wish I had called `bounding-box`)

Add a new field `polygon` that is a list of four (x, y) coordinates, taken from the raw locations - the point with lowest value of x, the highest value of x, the highest value of y, and the lowest value of y.

Barcodes and one-dimensional barcodes are all rectangular so I think that the four coordinates will be sufficient to describe polygon last around detected barcodes.
